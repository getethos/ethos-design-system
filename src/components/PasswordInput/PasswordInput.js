import React from 'react'
import { TextInput } from '../Inputs/TextInput'

export const PasswordInput = (props) => (
  <TextInput type="password" restrictIllegal={false} {...props} />
)
