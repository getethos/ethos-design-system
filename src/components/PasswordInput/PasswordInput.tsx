import * as React from 'react'
import { TextInput } from '../TextInput'
export const PasswordInput = (props) => (
  <TextInput type="password" restrictIllegal={false} {...props} />
)
