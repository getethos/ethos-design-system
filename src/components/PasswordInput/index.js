export { PasswordInput } from './PasswordInput.js'
