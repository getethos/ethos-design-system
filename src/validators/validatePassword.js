const passwordRegex = /^(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[!#$%^&*])[a-zA-Z0-9!#$%^&*]{8,100}$/

const PASSWORD_CREATION_MIN_LENGTH = 8
const PASSWORD_CREATION_MIN_LENGTH_ERROR_MESSAGE = `Password must contain at least ${PASSWORD_CREATION_MIN_LENGTH} characters and have 3 of the following: a number, lowercase letter, uppercase letter, and special character.`
const PASSWORD_CREATION_REQUIRED_ERROR_MESSAGE = PASSWORD_CREATION_MIN_LENGTH_ERROR_MESSAGE
const PASSWORD_VERIFICATION_REQUIRED_ERROR_MESSAGE = `Please enter your password.`

export default function validatePassword(x) {
  if (!x) return PASSWORD_VERIFICATION_REQUIRED_ERROR_MESSAGE 
  return passwordRegex.test(x) ? '' : PASSWORD_CREATION_REQUIRED_ERROR_MESSAGE 
}
