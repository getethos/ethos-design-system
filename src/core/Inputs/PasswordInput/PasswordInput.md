```jsx
import validatePassword from '../../../validators/validatePassword'
// formChangeHandler gets wired up automatically if using <Form /> component
const formChangeHandlerStub = () => {}
;<PasswordInput
  name="password-example"
  labelCopy="Password"
  data-tid="the-password-input"
  formChangeHandler={formChangeHandlerStub}
  validator={validatePassword}
/>
```
