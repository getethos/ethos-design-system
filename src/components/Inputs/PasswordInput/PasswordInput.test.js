import React from 'react'
import { PasswordInput } from './PasswordInput'
import renderer from 'react-test-renderer'
import validatePassword from '../../../validators/validatePassword'

const formChangeHandlerStub = jest.fn()

describe('PasswordInput', () => {
  it('default rendering', () => {
    const tree = renderer
      .create(
        <PasswordInput
          name="password-example"
          labelCopy="Password"
          data-tid="the-password-input"
          formChangeHandler={formChangeHandlerStub}
          validator={validatePassword}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
