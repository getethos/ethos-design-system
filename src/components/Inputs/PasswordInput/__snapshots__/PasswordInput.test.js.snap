// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput default rendering 1`] = `
Array [
  <label
    className="Caption Theinhardt Medium500 GrayPrimary AllCaps"
    htmlFor="password-example"
  >
    Password
  </label>,
  <div
    className="Spacer"
    style={
      Object {
        "height": 8,
      }
    }
  />,
  <input
    className="TextInput"
    data-tid="the-password-input"
    name="password-example"
    onBlur={[Function]}
    onChange={[Function]}
    onPaste={[Function]}
    type="password"
    value=""
  />,
  "",
]
`;
